### ISSUE TITLE
[cocom][colic] empty enriched indexes

### ISSUE THREAD

**Issue by GrayStranger** [chaoss/grimoirelab#598]
Hello, All

I can't get `cocom` and `colic` indexes filled.... no idea which way digging, but for now result is:
```
yellow open gitlab_merges_raw           dW5A7OkRQsSTOf2ERm-P7Q 5 1    103     5   2.9mb   2.9mb
yellow open .grimoirelab-sigils         aDhWnYcqTSCc7gMYO2I0fA 5 1     39     2    88kb    88kb
yellow open colic_chaoss                PvRbrZ3LRD-Ue6WL0Xmsvg 5 1   1189    87   1.4mb   1.4mb
yellow open git-aoc_enriched            9fUv9DYBQhm_FLcnLVuauA 5 1 272789 77217 323.9mb 323.9mb
yellow open git_raw                     9KX4FBhnTo2Ifhif79Nqow 5 1  20411     0  41.3mb  41.3mb
yellow open git_study_forecast_activity XLLiClXHQI2-g9t4NPL0KQ 5 1     65     7 225.5kb 225.5kb
yellow open cocom_chaoss                pm7uQm3sQwCNOjoC_NImrA 5 1   1189    75   2.7mb   2.7mb
yellow open colic_chaoss_enrich         q2lyh-kwT0ue3GuX8_iG4g 5 1      0     0   1.2kb   1.2kb
yellow open gitlab_merges_enriched      K9VVPrOmSUG0WVOA1qfJJw 5 1    103    30 369.9kb 369.9kb
yellow open jira_raw                    5ORCz2yJTSysSW_JluF4iQ 5 1  88754     1   1.9gb   1.9gb
yellow open cocom_enrich_graal_repo     0AnnohkNSiSiCJDeM7N83g 5 1      0     0   1.2kb   1.2kb
green  open .kibana_1                   j4McvhMOT2yti7675qYaqA 1 0    222    29 406.4kb 406.4kb
yellow open cocom_chaoss_enrich         KGNO_pbhQqyDsFPS8zAKbw 5 1      0     0   1.2kb   1.2kb
yellow open colic_enrich_graal_repo     jdEdlP5CSGqSs58hFN1uUQ 5 1      0     0   1.2kb   1.2kb
yellow open jira_enriched               PJSGOncaT12q2xElSkX8lQ 5 1 404486 90160   2.7gb   2.7gb
yellow open git_enriched                HVqoZeGnSza0kSl0a9EOOQ 5 1  20411  5618 107.8mb 107.8mb
yellow open git-onion_enriched          IvFrcGSVSjSmoQhs9kn42Q 5 1   1092     0 633.4kb 633.4kb
```

```
2023-05-29 21:04:56,585 - grimoire_elk.elk - INFO - [cocom] Starting study: enrich_cocom_analysis, params {}
2023-05-29 21:04:56,585 - grimoire_elk.enriched.cocom - INFO - [cocom] study enrich-cocom-analysis start
2023-05-29 21:04:56,590 - grimoire_elk.enriched.cocom - INFO - [cocom] study enrich-cocom-analysis 0 repositories to process
2023-05-29 21:04:56,669 - grimoire_elk.enriched.cocom - INFO - [cocom] study enrich-cocom-analysis End
```
and the same for `colic`

I had already tried delete indexes to let sirmordred re-create it - the same result. 

`cloc` and `fossology` builded inside the container with mordred and there are no related errors in log.

versions of all apps - from the last GrimoireLab 0.10.0 release.

HelpPlease :).

**Comment by zhquan** [comment]
Hi @FixItNowPlease 

Do you have any log related to the enrichment process (not the studies one)? Before `[cocom] Starting study` you should have
```
[cocom] enrichment phase start
...
...
[cocom] enrichment phase finished ...
```
Is there any error between these logs related to `cocom`?

**Comment by sduenas** [comment]
I hope I can have a look a it next week.

**Comment by zhquan** [comment]
Hi @GrayStranger 

Sorry for the late reply, I can create the `cocom` enriched index without any issue
- setup.cfg: you have to create first `/home/grimoire/worktrees/` in the mordred container or mount a volume with the correct permissions (for test you can set `chmod 777 <worktress_path>`)
```
[cocom]
raw_index = cocom_raw
enriched_index = cocom_enrich
category = code_complexity_lizard_file
studies = [enrich_cocom_analysis]
branches = master
worktree-path = /home/grimoire/worktrees/

[enrich_cocom_analysis]
out_index = cocom_chaoss_study
interval_months = [7]
```
- projects.json
```
{
  "GrimoireLab": {
    "cocom": [
      "https://github.com/chaoss/grimoirelab-toolkit"
    ]
  }
}
```
- all.logs
```
2023-07-19 09:42:10,980 - sirmordred.task_enrich - INFO - [cocom] enrichment phase starts
2023-07-19 09:42:11,062 - grimoire_elk.elastic - INFO - Created index https://localhost:9200/cocom_enrich
2023-07-19 09:42:11,068 - sirmordred.task_enrich - INFO - [cocom] enrichment starts for https://github.com/chaoss/grimoirelab-toolkit
2023-07-19 09:42:11,283 - grimoire_elk.elastic - INFO - Alias cocom created on https://localhost:9200/cocom_enrich.
2023-07-19 09:42:11,647 - grimoire_elk.enriched.cocom - INFO - [cocom] 287 items inserted
2023-07-19 09:42:11,648 - grimoire_elk.elk - INFO - [cocom] Done enrichment for https://github.com/chaoss/grimoirelab-toolkit
2023-07-19 09:42:11,648 - sirmordred.task_enrich - INFO - [cocom] enrichment finished for https://github.com/chaoss/grimoirelab-toolkit
2023-07-19 09:42:11,648 - sirmordred.task_enrich - INFO - [cocom] enrichment phase finished in 0:00:00
2023-07-19 09:42:11,648 - sirmordred.task_enrich - INFO - [cocom] data retention start
2023-07-19 09:42:11,665 - sirmordred.task_enrich - INFO - [cocom] data retention end
2023-07-19 09:42:11,665 - sirmordred.task_enrich - INFO - [cocom] identities retention end
2023-07-19 09:42:11,665 - sirmordred.task_enrich - INFO - [cocom] autorefresh start
2023-07-19 09:42:11,710 - sirmordred.task_enrich - INFO - [cocom] Refreshing identities
2023-07-19 09:42:11,757 - sirmordred.task_enrich - INFO - [cocom] autorefresh end
2023-07-19 09:42:11,758 - sirmordred.task_enrich - INFO - [cocom] studies phase start
2023-07-19 09:42:13,824 - sirmordred.task_enrich - INFO - [cocom] Executing studies ['enrich_cocom_analysis']
2023-07-19 09:42:13,825 - grimoire_elk.elk - INFO - [cocom] Starting study: enrich_cocom_analysis, params {'out_index': 'cocom_chaoss_study', 'interval_months': ['7']}
2023-07-19 09:42:13,826 - grimoire_elk.enriched.cocom - INFO - [cocom] study enrich-cocom-analysis start
2023-07-19 09:42:13,836 - grimoire_elk.enriched.cocom - INFO - [cocom] study enrich-cocom-analysis 1 repositories to process
2023-07-19 09:42:13,909 - grimoire_elk.elastic - INFO - Created index https://localhost:9200/cocom_chaoss_study
2023-07-19 09:42:13,966 - grimoire_elk.elastic - INFO - Alias cocom_study created on https://localhost:9200/cocom_chaoss_study.
2023-07-19 09:42:13,966 - grimoire_elk.enriched.cocom - INFO - [cocom] study enrich-cocom-analysis start analysis for https://github.com/chaoss/grimoirelab-toolkit
2023-07-19 09:42:14,121 - grimoire_elk.enriched.cocom - INFO - [cocom] study enrich-cocom-analysis 38 items inserted for Graal CoCom Analysis Study
2023-07-19 09:42:14,121 - grimoire_elk.enriched.cocom - INFO - [cocom] study enrich-cocom-analysis End analysis for https://github.com/chaoss/grimoirelab-toolkit with month interval
2023-07-19 09:42:14,121 - grimoire_elk.enriched.cocom - INFO - [cocom] study enrich-cocom-analysis End
2023-07-19 09:42:14,138 - sirmordred.task_enrich - INFO - [cocom] studies phase end
2023-07-19 09:42:14,138 - sirmordred.task_enrich - INFO - [cocom] autorefresh for studies start
2023-07-19 09:42:14,139 - sirmordred.task_enrich - INFO - [cocom] autorefresh for studies end
2023-07-19 09:42:14,139 - sirmordred.task_manager - INFO - [cocom] sleeping for 60 seconds
```
- indexes
```
GET _cat/indices?s=i

cocom_chaoss_study           L8dDQkVcQAe-T_0QRB6SAA 1 1      38      0  18.5kb  18.5kb
cocom_enrich                 xld4NZQDQU2nOcJIAmyDyw 1 1     287      0 274.1kb 274.1kb
cocom_raw                    jSJ99bUKQJiFH9bNx0XW4w 1 1     129      0 161.8kb 161.8kb
```

I'm using this version: `grimoirelab/grimoirelab:0.10.0`

**Comment by sduenas** [comment]
I think what we need it's the full list described in here: https://github.com/chaoss/grimoirelab-graal/#how-to-install-and-create-the-executables

**Comment by sduenas** [comment]
Hi @GrayStranger,

Is that gitlab repository public to your intranet? Does it require user and password to access the repo? Do the git indices contain any data?

PD: Sorry for our late response. Most of us are on vacation, so it's hard to be active here.

**Comment by sduenas** [comment]
That's weird. The difficult part is to get the raw indices. There must be some kind of error of exception we aren't seeing. Would you mind to set the platform to the debug mode and paste here the logs you get?

To do it, you have to set the `debug` parameter to `true` on the general section.

```
[general]
debug = true
...
```

Also, to reduce the noise and make the process faster, you should activate `cocom` or `colic` only. To do it, set to `false` everything under `phases` section but `enrichment` and comment the sections `git` and `gitlab`, studies, and `cocom` or `colic` - so we get debug messages from one of the processes only. This should look like:

```
[phases]
collection = false
identities = false
enrichment = true
panels = false
```

**Comment by zhquan** [comment]
Hi @GrayStranger 

Could you run this query in `dev tools` and paste the result? It seems that this query is returning 0 items
```
GET cocom_raw/_search
{
    "query": {
        "bool": {
            "filter": [
                {
                    "term": {
                        "origin": "https://git.domen.com/repo"
                    }
                }
            ]
        }
    },
    "sort": {
        "metadata__timestamp": {
            "order": "asc"
        }
    }
}
```

**Comment by zhquan** [comment]
> full format with creds inside origin stored in index, and default request to raw index with no creds within repo url - returns nothing. Am i right?

Yes, that means `cocom` and `colic` can only enrich repositories without creds in `projects.json`

**Comment by zhquan** [comment]
If you remove the creds in `projects.json` you have to remove it from all documents in the raw index (`origin` field) and also rename the clones.

For now, it only works for public repos

**Comment by zhquan** [comment]
The way to fix this is to anonymize the fields with credentials in the raw index so no credentials will be stored in the indexes, like https://github.com/chaoss/grimoirelab-elk/blob/master/grimoire_elk/raw/git.py#L75 

Cocom and colic use https://github.com/chaoss/grimoirelab-elk/blob/master/grimoire_elk/raw/graal.py

**Comment by zhquan** [comment]
If your code works, could you please create a PR with the fix?

**Comment by zhquan** [comment]
Fixed at chaoss/grimoirelab-elk#1114
